WPMedia/fusion-news-theme-blocks
# THEMES-987: Sanitize ANS

## Description
a sanitizeANS util method was added to engine-theme-sdk in 1.25 https://github.com/WPMedia/engine-theme-sdk/blob/arc-themes-release-version-1.25/src/utils/sanitizeANS.ts
The following content sources now use a sanitizeANS method in their transforms to check for known internal fields to empty out:
* @wpmedia/collections-content-source-block
* @wpmedia/content-api-source-block
* @wpmedia/story-feed-author-content-source-block
* @wpmedia/story-feed-query-content-source-block
* @wpmedia/story-feed-sections-content-source-block
* @wpmedia/story-feed-tag-content-source-block


### Note on test coverage
There is no change to test cov, the method has coverage in the engine-theme-sdk. However, since I added a method the poor coverage is now showing. Given that we are moving to deprecate 1.x and there is no change here, the recommendation is to merge without increasing coverage.

<img width="1127" alt="image" src="https://user-images.githubusercontent.com/2506649/217268901-ac70d450-5e62-4ae3-814f-2f4404704787.png">



## Jira Ticket

- [THEMES-987](https://arcpublishing.atlassian.net/browse/THEMES-987)

## Acceptance Criteria


## Test Steps

- Add test steps a reviewer must complete to test this PR

1. Checkout this branch `git checkout THEMES-987`
2. Run fusion repo with linked blocks `npx fusion start -f -l @wpmedia/collections-content-source-block,@wpmedia/content-api-source-block,@wpmedia/story-feed-author-content-source-block,@wpmedia/story-feed-query-content-source-block,@wpmedia/story-feed-sections-content-source-block,@wpmedia/story-feed-tag-content-source-block`
    - Related feature pack blocks.json 1.25 bump branch/PR - https://github.com/WPMedia/arc-themes-feature-pack/pull/383
3. Create a story in composer that populates any of the fields 
    - e.g., https://sandbox.themesinternal.arcpublishing.com/composer/edit/ELDHUK7DONCL5NYNMGS6AZ3MP4/
    - story ID: ELDHUK7DONCL5NYNMGS6AZ3MP4
    - story path: /news/2023/02/04/internal-fields/
4. On current arc-themes-release-version-1.25, check the content-api source with `the-gazette` and the story above to confirm you do see data returning in one of the following fields
```
 * - editor_note
 * - planning.internal_note
 * - workflow
 * - additional_properties.clipboard
 * - content_elements.additional_properties.comments
 * - content_elements.additional_properties.inline_comments
 ```
5. Use the content source debugger for every content source listed in the description and confirm any of those fields do not populate any data but it returns the same data type with an empty string, object or array as if it was never populated.
  - Note: you don't need to test every field necessarily here, we're mostly looking to ensure the content source still returns as expected, and the sanitizeANS method is functioning. It's likely the exact fields will change in the sdk package but it's not a concern within an arc-themes-block content source, ownership for that logic is in engine-theme-sdk.

## Effect Of Changes

### Before
content-api and content-feed content sources return all ANS fields

### After
content-api and content-feed content sources return sanitized ANS fields

 
## Dependencies or Side Effects
engine-theme-sdk 1.25 includes the sanitizeANS method that this change depends on.


## Author Checklist

_The author of the PR should fill out the following sections to ensure this PR is ready for review._

        please explain why (so that we can fix it whenever it gets refactored).

## Reviewer Checklist

_The reviewer of the PR should copy-paste this template into the review comments on review._



[THEMES-987]: https://arcpublishing.atlassian.net/browse/THEMES-987?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
